app/src/main/java/com/leoart/koreanphrasebook/ui/vocabulary/DictionaryAdapter.kt

```diff
@@ -91,16 +99,24 @@ class DictionaryAdapter(dictionary: Dictionary) : SectioningAdapter(), FastScrol
     }
 
     override fun getSectionName(position: Int): String {
-        if (letters == null || letters.size < position) {
-            return ""
+        letters?.let {
+            if (it.size >= position)
+                return it[position].toString()
         }
-        return letters[position].toString()
+        return ""
     }
 
     class DictViewHolder internal constructor(itemView: View) : SectioningAdapter.ItemViewHolder(itemView) {
         var text = itemView.findViewById<TextView>(R.id.text)
     }
 
+    fun setData(dictionary: Dictionary) {
```

Comment:
@greenhorn31 why is this done on update gradle branch? You should separate tasks in branches